lib/robots/dor_repo/assembly/content_metadata_create.rb

```diff
@@ -11,19 +11,35 @@ def initialize
         end
 
         # Generate the structural metadata for this object from stub content metadata (if present).
+        # rubocop:disable Metrics/MethodLength
         def perform_work
           return LyberCore::ReturnState.new(status: :skipped, note: 'object is not an item') unless assembly_item.item? # not an item, skip
 
           return LyberCore::ReturnState.new(status: :skipped, note: 'No stubContentMetadata to load from the filesystem') unless assembly_item.stub_content_metadata_exists?
 
-          updated = assembly_item.cocina_model.new(structural: assembly_item.convert_stub_content_metadata)
-          assembly_item.object_client.update(params: updated)
+          tries = 0
+          max_tries = 3
+          begin
+            updated = assembly_item.cocina_model.new(structural: assembly_item.convert_stub_content_metadata)
+            assembly_item.object_client.update(params: updated)
+          # sometimes the stubContentMetadata.xml is in process of being written and is invalid
+          # this is a workaround that gives it more time to complete
+          # see https://github.com/sul-dlss/common-accessioning/issues/1477
+          rescue RuntimeError => e
```

Comment:
Is this the most specific error that you can catch?

Reply:
Currently we just raise a generic RuntimeError when there is a problem parsing the stub content metadata: https://github.com/sul-dlss/common-accessioning/blob/main/lib/dor/assembly/stub_content_metadata_parser.rb#L47 

We could create a custom exception and then rescue only that, though seems potentially overkill (and we also just raise runtime errors elsewhere in this codebase).  Worst case is we rescue something else we didn't really intend to, retry a couple times and then raise it again anyway and fail.